### PR TITLE
Completed validation

### PR DESCRIPTION
**Pull request by BobbyMcWho**
@JLHOLM
@jaybobo 
Decided to try this out in coffeescript, not sure if I used it correctly or not but it compiles correctly and functions.

### CODE REVIEW

File: validator.coffee [hunk 1]
```diff
@@ -0,0 +1,28 @@
+$(document).ready(() ->
+  $('.errors').children().hide()
+)
+
+$('input[type="submit"').click((e) ->
+  e.preventDefault()
+  $('.errors').children().hide()
+  emailValid = checkEmail($('[placeholder="Email"]').val())
+  passwordValid = checkPassword($('[placeholder="Password"]').val())
```
Comment (sjreich) [inline validator.coffee:9]: It's a bit weird to be grabbing these elements by the placeholder, although I suppose you aren't allowed to change the html.  I guess ideally they'd have an id.

File: validator.coffee [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  if  emailValid && passwordValid
+    $('form').submit()
+    alert 'Form submitted'
+)
+
+checkEmail = (input) -> 
+  if !/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/g.test(input)
```
Comment (sjreich) [inline validator.coffee]: If I were doing this, I think I'd want to break the regex out into a constant.

File: validator.coffee [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    $('.errors li:nth-child(1)').show()
+    return false
+  else
+    return true
+
+checkPassword = (input) ->
+  if !/^(?=.*\d)(?=.*[A-Z]).{8,}$/g.test(input)
+    $('.errors li:nth-child(2), .errors li:nth-child(3), .errors li:nth-child(4)').show()
```
Comment (sjreich) [inline validator.coffee]: It looks like this isn't going to allow you to distinguish among the various ways the password could go wrong - if there's any problem, you display all of the errors.  Is that intended?

Comment (sjreich) [reply validator.coffee]: Also, I think I'd want id's for these error messages.  (I know you aren't allowed to touch the html.)

File: validator.coffee [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    return false
+  else
+    return true
```
Comment (sjreich) [inline validator.coffee]: If you flip the order on these conditionals, you can avoid the `!`, which might improve readability a bit.

File: validator.coffee (hunk 3 above)
Comment (sjreich) [reply validator.coffee]: I like the lookaheads!
Not sure if the `^` and `$` are getting you anything in this case.

**Comment by sjreich** [comment]
I haven't thought through this one in a while.  Forgive me for going through it super carefully.

File: validator.coffee (hunk 1 above)
Comment (BobbyMcWho) [reply validator.coffee:9]: I agree, the restriction was the only reason I didn't add id's in.

File: validator.coffee (hunk 2 above)
Comment (BobbyMcWho) [reply validator.coffee]: Probably a good idea, if I had used it in more than one spot I definitely would have.

**Comment by BobbyMcWho** [comment]
@sjreich I adjusted some things based on your feedback

**Pull request closed without merging**